presto-main/src/main/java/com/facebook/presto/operator/scalar/ScalarFunctionImplementation.java

```diff
@@ -99,6 +100,17 @@ public ScalarFunctionImplementation(
             checkArgument(instanceType.equals(methodHandle.type().parameterType(0)), "methodHandle is not an instance method");
         }
 
+        List<Class<?>> parameterList = methodHandle.type().parameterList();
+        if (parameterList.contains(ConnectorSession.class)) {
+            checkArgument(parameterList.stream().filter(ConnectorSession.class::equals).count() <= 1, "function implementation cannot have more than 1 ConnectorSession argument");
```

Comment:
if `parameterList.contains(ConnectorSession.class)` we should have exactly one `ConnectorSession` parameter so `checkArgument` on L105 should check for equality (parameterList.stream().filter(ConnectorSession.class::equals).count() == 1).

You can then update the error message too.

Reply:
Nice catch!! Thank you !